packages/domain-picker/src/hooks/use-track-modal.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useOnUnmount } from './use-on-unmount';
+
+/**
+ * Records events in tracks on opening and closing the modal
+ * When closing the modal, additional properties can be recorded e.g. which domain was selected.
+ *
+ * @param modalName The name of the modal as will be sent to tracks
+ * @param onModalOpen called when the modal opens
+ * @param onModalUnmount called when the modal unmounts
+ * @param eventProperties Returns additional properties to be recorded on completing the modal
+ */
+export function useTrackModal(
+	modalName: string,
+	onModalOpen: Function,
+	onModalUnmount: Function,
+	eventProperties: { selected_domain?: string }
```

Comment:
`useTrackModal` seems like a generic analytics tool for tracking modals so I guess it fits in the category mentioned in https://github.com/Automattic/wp-calypso/pull/42409/files#r428674134
Not a blocker now but we should extract before doing the exact same thing for the PlansGrid.